module/VuFind/src/VuFind/Hierarchy/TreeRenderer/HTMLTree.php

```diff
@@ -0,0 +1,267 @@
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:hierarchy_components Wiki
+ */
+
+namespace VuFind\Hierarchy\TreeRenderer;
+
+use Laminas\Mvc\Controller\Plugin\Url as UrlPlugin;
+use Laminas\View\Renderer\RendererInterface;
+
+use function in_array;
+
+/**
+ * Hierarchy Tree HTML Renderer
+ *
+ * This is a helper class for producing hierarchy trees.
+ *
+ * @category VuFind
+ * @package  HierarchyTree_Renderer
+ * @author   Luke O'Sullivan <l.osullivan@swansea.ac.uk>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:hierarchy_components Wiki
+ */
+class HTMLTree extends AbstractBase implements \VuFind\I18n\Translator\TranslatorAwareInterface
+{
+    use \VuFind\I18n\Translator\TranslatorAwareTrait;
+
+    /**
+     * Router plugin
+     *
+     * @var UrlPlugin
+     */
+    protected $router = null;
+
+    /**
+     * Whether the collections functionality is enabled
+     *
+     * @var bool
+     */
+    protected $collectionsEnabled;
+
+    /**
+     * View renderer
+     *
+     * @var RendererInterface
+     */
+    protected $viewRenderer;
+
+    /**
+     * Constructor
+     *
+     * @param UrlPlugin         $router             Router plugin for urls
+     * @param bool              $collectionsEnabled Whether the collections functionality is enabled
+     * @param RendererInterface $renderer           View renderer
+     */
+    public function __construct(UrlPlugin $router, bool $collectionsEnabled, RendererInterface $renderer)
+    {
+        $this->router = $router;
+        $this->collectionsEnabled = $collectionsEnabled;
+        $this->viewRenderer = $renderer;
+    }
+
+    /**
+     * Get a list of trees containing the item represented by the stored record
+     * driver.
+     *
+     * @param string $hierarchyID Optional filter: specific hierarchy ID to retrieve
+     *
+     * @return mixed An array of hierarchy IDS if an archive tree exists,
+     * false if it does not
+     */
+    public function getTreeList($hierarchyID = false)
+    {
+        $record = $this->getRecordDriver();
+        $inHierarchies = $record->getHierarchyTopID();
+        $inHierarchiesTitle = $record->getHierarchyTopTitle();
+
+        if ($hierarchyID) {
+            // Specific Hierarchy Supplied
+            if (
+                in_array($hierarchyID, $inHierarchies)
+                && $this->getDataSource()->supports($hierarchyID)
+            ) {
+                return [
+                    $hierarchyID => $this->getHierarchyName(
+                        $hierarchyID,
+                        $inHierarchies,
+                        $inHierarchiesTitle
+                    ),
+                ];
+            }
+        } else {
+            // Return All Hierarchies
+            $i = 0;
```

Comment:
Is there a reason to initialize and increment `$i` separately? Couldn't we just use `foreach ($inHierarchies as $i => $hierarchyTopID) {` below to achieve the same effect? I think it's safe to assume that $inHierarchies and $inHierarchiesTitle have parallel keys.

Reply:
It's just copy-paste from JSTree.php, but looks like you're right. Now changed.